tests/functional/test_auth_config.py

```diff
@@ -75,3 +75,15 @@ def assert_all_requirements_match(auth_config, message):
         AUTH_TYPE_REQUIREMENTS[auth_type] for auth_type in auth_config
     )
     assert len(auth_requirements) == 1
+
+
+def test_sigv4a_signing_region_set_config_from_environment():
```

Comment:
It may be worth parameterizing this so we're testing environment, set Config, and a default Config to exercise all your code paths in `_compute_sigv4a_signing_region_set_config`.